bin/odyssey/src/main.rs

```diff
@@ -70,11 +72,23 @@ fn main() {
                             .collect::<Result<_, _>>()
                             .wrap_err("No valid EXP0001 delegations specified")?;
 
+                        // construct a boxed rpc client
+                        let rpc_client = RpcClient::new_http(
+                            format!(
+                                "http://{}:{}",
+                                ctx.config().rpc.http_addr,
+                                ctx.config().rpc.http_port
+                            )
+                            .parse()
+                            .expect("invalid rpc url, this should not happen"),
+                        )
+                        .boxed();
                         ctx.modules.merge_configured(
                             OdysseyWallet::new(
-                                ctx.provider().clone(),
-                                wallet,
-                                ctx.registry.eth_api().clone(),
+                                ProviderBuilder::new()
+                                    .with_recommended_fillers()
+                                    .wallet(wallet)
+                                    .on_client(rpc_client),
                                 ctx.config().chain.chain().id(),
```

Comment:
we could get rid of the chain id arg if we weren't dependent on integrating it into the sequencer also; in that case, the service could just fetch the chain id. but since we init the module here before the rpc server has started, this is not possible